framework/source/class/qx/core/Object.js

```diff
@@ -230,15 +230,25 @@ qx.Class.define("qx.core.Object",
     /**
      * Load user defined data from the object
      *
-     * @param key {String} the key
+     * @param key {String?} the key
+     *   If undefined, pass the user a copy of the entire map
```

Comment:
It's likely that someone would prefer to pass `null` or `false` here to explicitly indicate that the entire map is requested. The code already allows that. The documentation should probably say something like, `If not a string, return a copy of the entire map.`

Reply:
Or better yet, `If falsy, return a copy of the entire map."

Reply:
I am concerned that someone might use something that is falsy as a key in the map. Consider
```
obj.setUserData(0, 'test');
```

Reply:
Now that I've changed that code I think it was better before. In the setUserData() function there is nothing to require that the keys of the map are strings.

I posit that the check should be `if (typeof key === "undefined") ... return map`

Reply:
I would venture that that's an oversight in the code, as the documented behavior is that the key must be a string... and using a number as a key in a map implicitly converts it to a string.

Well in any case, how about we at least allow `undefined` or `null` to mean return the map (and document it as such)?

Reply:
I believe this has been resolved in the latest changes.